pkg/catalog/uncommitted_iterator.go

```diff
@@ -0,0 +1,133 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/pkg/graveler"
+)
+
+type UncommittedIterator struct {
+	store     Store
+	ctx       context.Context
+	err       error
+	branchItr graveler.BranchIterator
+	entryItr  *valueEntryIterator
+	entry     *UncommittedRecord
+}
+
+type UncommittedRecord struct {
+	branchID graveler.BranchID
+	*EntryRecord
+}
+
+func NewUncommittedIterator(ctx context.Context, store Store, repository *graveler.RepositoryRecord) (*UncommittedIterator, error) {
+	bItr, err := store.ListBranches(ctx, repository)
+	if err != nil {
+		return nil, err
+	}
+	return &UncommittedIterator{
+		store:     store,
+		ctx:       ctx,
+		branchItr: bItr,
+	}, nil
+}
+
+// nextStaging reads the next branch staging area into entryItr
+func (u *UncommittedIterator) nextStaging() bool {
+	if u.entryItr != nil {
+		u.entryItr.Close()
```

Comment:
verify that multiple calls to Close are handled by the iterator or use nil as a marker